lib/v2/screens/profile_screens/profile/interactor/viewmodels/profile_event.dart

```diff
@@ -45,3 +45,15 @@ class OnCurrencyChanged extends ProfileEvent {
   @override
   String toString() => 'OnCurrencyChanged';
 }
+
+class ShowLogoutButton extends ProfileEvent {
```

Comment:
take a look at PageCommand

follow the page command pattern.

Reply:
So you want me to use a listener instead of this onTap?

Reply:
So.. yes a listener and the pattern